### PR TITLE
test(robot): Add test cases for Test Volume Attached at Maximum Snapshot Count

### PR DESCRIPTION
**Pull request by roger-ryao**
#### Which issue(s) this PR fixes:
longhorn/longhorn#10322

#### What this PR does / why we need it:
Validate that reaching the snapshot limit of 250 does not affect volume attach/detach operations.

#### Special notes for your reviewer:
@PhanLe1010 

#### Additional documentation or context
longhorn/longhorn#10322
longhorn/longhorn#10308

- PR test log : 
  - [pr2312log.tar.gz](https://github.com/user-attachments/files/18904572/pr2312log.tar.gz)

- Here's how to execute test case: ` ./run.sh -t 'Test Volume Attached at Maximum Snapshot Count'`
Screenshot_20250221_175411

## Summary by CodeRabbit

- **New Features**
	- Enhanced snapshot operations for greater consistency and reliability during creation and deletion.
	- Introduced a new feature for batch creation of snapshots.
- **Tests**
	- Introduced an end-to-end validation to ensure volume attach and detach operations remain stable when approaching the maximum snapshot limit.
	- Added a new test case to validate functionality when the maximum snapshot count is reached.

### CODE REVIEW

**Comment by derekbit** [comment]
Do we have e2e test cases for the setting `snapshot-max-count`?

**Comment by roger-ryao** [comment]
> Do we have e2e test cases for the setting `snapshot-max-count`?

Hi @derekbit 
    I searched for the keyword `snapshot-max-count` in the `longhorn-tests` but did not find any related test cases.

File: e2e/tests/regression/test_volume.robot [hunk 1]
```diff
@@ -65,3 +66,18 @@ Test RWX Volume Without Migratable Should Be Incompatible With Strict-Local
...
+    [Tags]    volume    snapshot-limit
+    [Documentation]    Validate that reaching the snapshot limit of 250 does not affect
+    ...                volume attach/detach operations.
+    ...                Issue: https://github.com/longhorn/longhorn/issues/10308
+    Given Create volume 0 with    dataEngine=${DATA_ENGINE}    numberOfReplicas=1
+    When Attach volume 0
+    And Wait for volume 0 healthy
+    FOR    ${i}    IN RANGE    249
+        Create snapshot ${i} of volume 0
+    END
```
Comment (c3y1huang) [inline e2e/tests/regression/test_volume.robot:80]: Suggest to wrap this in a keyword, e.g, `Create ${snapshot_count} snapshot for volume ${volume_id}`

File: e2e/tests/regression/test_volume.robot [hunk 2]
```diff
@@ -65,3 +66,18 @@ Test RWX Volume Without Migratable Should Be Incompatible With Strict-Local
...
+    And Detach volume 0
+    When Attach volume 0
+    And Wait for volume 0 healthy
```
Comment (c3y1huang) [inline e2e/tests/regression/test_volume.robot:83]:
```suggestion
    Given Create volume 0 with    dataEngine=${DATA_ENGINE}    numberOfReplicas=1
    And Attach volume 0
    And Wait for volume 0 healthy
    When Create 249 snapshot for volume 0
    And Detach volume 0
    Then Attach volume 0
    And Wait for volume 0 healthy
```

File: e2e/keywords/snapshot.resource [hunk 3]
```diff
@@ -55,3 +56,8 @@ Validate snapshot ${snapshot_id} is in volume ${volume_id} snapshot list
 Create snapshot ${snapshot_id} of volume ${volume_id} will fail
     ${volume_name} =    generate_name_with_suffix    volume    ${volume_id}
     Run Keyword And Expect Error    *    create_snapshot    ${volume_name}    ${snapshot_id}    waiting=False
+
+Create ${snapshot_count} snapshot for volume ${volume_id}
+    FOR    ${i}    IN RANGE    ${snapshot_count}
+        Create snapshot ${i} of volume ${volume_id}
+    END
```
Comment (c3y1huang) [inline e2e/keywords/snapshot.resource:63]: Add an EOF blank line.

**Review by c3y1huang — CHANGES_REQUESTED** [review]

**Review by c3y1huang — APPROVED** [review]

**Comment by yangchiu** [comment]
@mergify backport v1.8.x
